crates/types/src/simple_vote.rs

```diff
@@ -0,0 +1,225 @@
+    /// Commitment to the leaf
+    pub leaf_commit: Commitment<LEAF>,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a DA vote.
+pub struct DAData<BLOCK: Committable> {
+    /// Commitment to a block
+    pub block_commit: Commitment<BLOCK>,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a timeout vote.
+pub struct TimeoutData<TYPES: NodeType> {
+    /// View the timeout is for
+    pub view: TYPES::Time,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a VID vote.
+pub struct VIDData<BLOCK: Committable> {
+    /// Commitment to the block the VID vote is on.
+    pub block_commit: Commitment<BLOCK>,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a Pre Commit vote.
+pub struct ViewSyncPreCommitData<TYPES: NodeType> {
+    /// The relay this vote is intended for
+    pub relay: EncodedPublicKey,
+    /// The view number we are trying to sync on
+    pub round: TYPES::Time,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a Commit vote.
+pub struct ViewSyncCommitData<TYPES: NodeType> {
+    /// The relay this vote is intended for
+    pub relay: EncodedPublicKey,
+    /// The view number we are trying to sync on
+    pub round: TYPES::Time,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a Finalize vote.
+pub struct ViewSyncFinalizeData<TYPES: NodeType> {
+    /// The relay this vote is intended for
+    pub relay: EncodedPublicKey,
+    /// The view number we are trying to sync on
+    pub round: TYPES::Time,
+}
+
+/// Marker trait for data or commitments that can be voted on.
+/// Only structs in this file can implement voteable.  This is enforced with the `Sealed` trait
+/// Sealing this trait prevents creating new vote types outside this file.
+pub trait Voteable:
+    sealed::Sealed + Committable + Clone + Serialize + Debug + PartialEq + Hash + Eq
+{
+}
+
+/// Sealed is used to make sure no other files can implement the Voteable trait.
+/// All simple voteable types should be implemented here.  This prevents us from
+/// creating/using improper types when using the vote types.
+mod sealed {
+    use commit::Committable;
+
+    /// Only structs in this file can impl `Sealed`
+    pub trait Sealed {}
+
+    // TODO: Does the implement for things outside this file that are commitable?
+    impl<C: Committable> Sealed for C {}
+}
+
+/// A simple yes vote over some votable type.  
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+pub struct SimpleVote<TYPES: NodeType, DATA: Voteable, MEMBERSHIP: Membership<TYPES>> {
+    /// The signature share associated with this vote
+    pub signature: (EncodedPublicKey, EncodedSignature),
+    /// The leaf commitment being voted on.
+    pub data: DATA,
+    /// The view this vote was cast for
+    pub current_view: TYPES::Time,
```

Comment:
Nit: We should change current_view to just view_number, since "current_view" isn't the best descriptor.